src/pal/src/exception/signal.cpp

```diff
@@ -554,6 +555,23 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
         ASSERT("sigprocmask failed; error is %d (%s)\n",errno, strerror(errno));
     } 
 
+#if defined(_AMD64_)
+    if (pointers->ExceptionRecord->ExceptionCode == EXCEPTION_INT_DIVIDE_BY_ZERO)
+    {
+        // It is possible that an overflow may also have been mapped to a divide-by-zero
+        // exception. This happens when we try to divide the maximum negative value of a
+        // signed integer with -1. 
+        //
+        // Thus, we will attempt to decode the instruction @ RIP to determine if that
+        // is the case using the faulting context.
+        if (IsDivByZeroAnIntegerOverflow(pointers->ContextRecord, pointers->ExceptionRecord))
```

Comment:
I can move it to coreclr, I would put it right into the `HandleHardwareException` in that case. But then the debugger would need to call the same code in its hardware exception handler.
Trying to consider the potential dangers, it seems that the only thing that can happen is that some other thread unmaps the memory where the operand is stored before we try to read it. In that case, we would get a double fault and crash. But if someone unmapped the memory a fraction of a second before the IDIV was executed, a SIGSEGV would occur in the native code and tear down the process anyways. So there doesn't seem to be much of a difference.